components/db/events.go

```diff
@@ -0,0 +1,247 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/leptonai/gpud/log"
+	"github.com/leptonai/gpud/pkg/sqlite"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+const (
+	// Event timestamp in unix seconds.
+	ColumnTimestamp = "timestamp"
+
+	// e.g., "dmesg", "nvml", "nvidia-smi".
+	ColumnDataSource = "data_source"
+
+	// e.g., "memory_oom", "memory_oom_kill_constraint", "memory_oom_cgroup", "memory_edac_correctable_errors".
+	ColumnEventType = "event_type"
+
+	// e.g., "xid", "sxid".
+	ColumnEventID1 = "event_id_1"
+
+	// e.g., "gpu_id", "gpu_uuid".
+	ColumnEventID2 = "event_id_2"
+
+	// e.g., "oom_reaper: reaped process 345646 (vector), now anon-rss:0kB, file-rss:0kB, shmem-rss:0".
+	ColumnEventDetails = "event_details"
+)
+
+type Event struct {
+	// unix seconds
+	Timestamp    int64  `json:"timestamp"`
+	DataSource   string `json:"data_source"`
+	EventType    string `json:"event_type"`
+	EventID1     string `json:"event_id_1"`
+	EventID2     string `json:"event_id_2"`
+	EventDetails string `json:"event_details"`
+}
+
+type DB struct {
+	table string
+	dbRW  *sql.DB
+	dbRO  *sql.DB
+}
+
+// Creates a new DB instance with the table created.
+func NewDB(ctx context.Context, dbRW *sql.DB, dbRO *sql.DB, tableName string) (*DB, error) {
+	if err := createTable(ctx, dbRW, tableName); err != nil {
+		return nil, err
+	}
+	return &DB{
+		table: tableName,
+		dbRW:  dbRW,
+		dbRO:  dbRO,
+	}, nil
+}
+
+func (db *DB) Insert(ctx context.Context, ev Event) error {
+	return insertEvent(ctx, db.dbRW, db.table, ev)
```

Comment:
do we allow one of dbRW and dbRO to be nil in the NewDB function? if yes, we should check nil before calling insertEvent